NohBoard/Extra/TRectangle.cs

```diff
@@ -0,0 +1,137 @@
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="TRectangle" /> instance from a position and size.
+        /// </summary>
+        public TRectangle(TPoint position, TPoint size)
+        {
+            this.Left = position.X;
+            this.Top = position.Y;
+            this.Right = position.X + size.X;
+            this.Bottom = position.Y + size.Y;
+        }
+
+        #endregion Constructors
+
+        #region Edges
+
+        /// <summary>
+        /// X-coordinate of the left edge
+        /// </summary>
+        public int Left { get; private set; }
+
+        /// <summary>
+        /// Y-coordinate of the top edge
+        /// </summary>
+        public int Top { get; private set; }
+
+        /// <summary>
+        /// X-coordinate of the rightedge
+        /// </summary>
+        public int Right { get; private set; }
+
+        /// <summary>
+        /// Y-coordinate of the bottom edge
+        /// </summary>
+        public int Bottom { get; private set; }
+
+        #endregion Edges
+
+        #region Corner points
+
+        /// <summary>
+        /// Alias for TopLeft.
+        /// </summary>
+        public TPoint Position => this.TopLeft;
+
+        /// <summary>
+        /// Returns a Size object with the Width and Height properties of this rectangle.
+        /// </summary>
+        public Size Size =>new Size(this.Right - this.Left, this.Bottom - this.Top);
+
+        /// <summary>
+        /// Returns the top left corner of the rectangle as a TPoint.
+        /// </summary>
+        public TPoint TopLeft => new TPoint(this.Left, this.Top);
+
+        /// <summary>
+        /// Returns the top right corner of the rectangle as a TPoint.
+        /// </summary>
+        public TPoint TopRight => new TPoint(this.Right, this.Top);
+
+        /// <summary>
+        /// Returns the bottom left corner of the rectangle as a TPoint.
+        /// </summary>
+        public TPoint BottomLeft => new TPoint(this.Left, this.Bottom);
+
+        /// <summary>
+        /// Returns the bottom right corner of the rectangle as a TPoint.
+        /// </summary>
+        public TPoint BottomRight => new TPoint(this.Right, this.Bottom);
+
+        #endregion Corner points
+
+        #region Methods
+
+        /// <summary>
+        /// Creates a rectangle that surrounds all points in a list.
+        /// </summary>
+        public static TRectangle FromPointList(TPoint[] points)
+        {
+            var left = int.MaxValue;
+            var top = int.MaxValue;
+            var right = int.MinValue;
+            var bottom = int.MinValue;
+
+            foreach (var point in points)
+            {
+                left = Math.Min(left, point.X);
+                top = Math.Min(top, point.Y);
+                right = Math.Max(right, point.X);
+                bottom = Math.Max(bottom, point.Y);
+            }
+
+            return new TRectangle(left, top, right, bottom);
+        }
+
+        #endregion
```

Comment:
Please document endregions as wel, for consistency with the rest. Though I feel for a single method, a region isn't particularly useful anyway.

Reply:
Oops. That one slipped through. My bad.

The reason I added a region is because more methods could be added in the future. Also, I like to be able to fold *everything* except what I'm currently working on. If you don't like it and if it's against the style you're using in this project, I'll remove it, though.